tests/legacy/generate_checkpoints.sh

```diff
@@ -1,44 +1,57 @@
 #!/bin/bash
-# Sample call:
-#  bash generate_checkpoints.sh 1.0.2 1.0.3 1.0.4
-
+# Usage:
+# 1. Generate checkpoints with one or more specified PL versions:
+#    bash generate_checkpoints.sh 1.0.2 1.0.3 1.0.4
+# 2. Generate checkpoints with the PL version installed in your environment:
+#    bash generate_checkpoints.sh
 set -e
 
-LEGACY_PATH="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
-FROZEN_MIN_PT_VERSION="1.6"
-
-echo $LEGACY_PATH
-# install some PT version here so it does not need to reinstalled for each env
-pip install virtualenv "torch==1.6" --quiet
-
-ENV_PATH="$LEGACY_PATH/vEnv"
+LEGACY_PATH=$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )
+ENV_PATH=$LEGACY_PATH/vEnv
+echo LEGACY_PATH: $LEGACY_PATH
+echo ENV_PATH: $ENV_PATH
 
 # iterate over all arguments assuming that each argument is version
-for ver in "$@"
+for pl_ver in "$@"
 do
-	echo "processing version: $ver"
-	# mkdir "$LEGACY_PATH/$ver"
-
-  # create local env
-  echo $ENV_PATH
-  virtualenv $ENV_PATH --system-site-packages
-  # activate and install PL version
-  source "$ENV_PATH/bin/activate"
+	echo processing version: $pl_ver
+
+  # Don't install/update anything before activating venv
+  # to avoid breaking any existing environment.
+  python -m venv $ENV_PATH
+  source $ENV_PATH/bin/activate
+
   # there are problem to load ckpt in older versions since they are saved the newer versions
-  pip install "pytorch_lightning==$ver" "torch==$FROZEN_MIN_PT_VERSION" "torchmetrics" "scikit-learn" --quiet
+  python -m pip install pytorch_lightning==$pl_ver -r $LEGACY_PATH/requirements.txt -f https://download.pytorch.org/whl/cpu/torch_stable.html
 
   python --version
-  pip --version
-  pip list | grep torch
+  python -m pip --version
+  python -m pip list | grep -e torch -e learn
 
-  python "$LEGACY_PATH/simple_classif_training.py"
-  cp "$LEGACY_PATH/simple_classif_training.py" ${LEGACY_PATH}/checkpoints/${ver}
+  python $LEGACY_PATH/simple_classif_training.py  # > /dev/null 2>&1
 
-  mv ${LEGACY_PATH}/checkpoints/${ver}/lightning_logs/version_0/checkpoints/*.ckpt ${LEGACY_PATH}/checkpoints/${ver}/
-  rm -rf ${LEGACY_PATH}/checkpoints/${ver}/lightning_logs
+  cp $LEGACY_PATH/simple_classif_training.py $LEGACY_PATH/checkpoints/$pl_ver
+  mv $LEGACY_PATH/checkpoints/$pl_ver/lightning_logs/version_0/checkpoints/*.ckpt $LEGACY_PATH/checkpoints/$pl_ver/
+  rm -rf $LEGACY_PATH/checkpoints/$pl_ver/lightning_logs
 
   deactivate
-  # clear env
   rm -rf $ENV_PATH
-
 done
+
+# use the PL installed in the environment if no PL version is specified
+if [[ -z "$@" ]]; then
+  pl_ver=$(python -c "import pytorch_lightning as pl; print(pl.__version__)")
+	echo processing version: $pl_ver
+
+  python -m pip install -r $LEGACY_PATH/requirements.txt
+
+  python --version
```

Comment:
Is this duplicated because the above is in a loop? You can define a function to reuse the code

Reply:
Done!